gui/create-item.lua

```diff
@@ -255,15 +259,38 @@ function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, ge
   item_type = "GLOB"
  elseif CORPSE_PIECES[layerName] or HAIR_PIECES[layerName] then -- check if hair
   item_type = "CORPSEPIECE"
+ elseif LIQUID_PIECES[layerName] then -- check if hair
+  item_type = "LIQUID_MISC" -- prolly need to do more work on this (it doesn't like spawning LIQUID items), but it works for now
+  liquid = true
+ end
+ if isCorpse then
+  item_type = "CORPSE"
+  generic = true
  end
  local itemType = dfhack.items.findType(item_type..":NONE")
  local itemSubtype = dfhack.items.findSubtype(item_type..":NONE")
  local material = "CREATURE_MAT:"..raceName..":"..layerMat
  local materialInfo = dfhack.matinfo.find(material)
  local item_id = dfhack.items.createItem(itemType, itemSubtype, materialInfo['type'], materialInfo.index, creator)
  local item = df.item.find(item_id)
- item:setQuality(quality)
- local isCorpsePiece = false
+ if liquid then
+  local bucketMat = dfhack.matinfo.find("PLANT_MAT:NETHER_CAP:WOOD")
+  if not bucketMat then
+   for i,n in ipairs(df.global.world.raws.plants.all) do
+    if n.flags.TREE then
+     bucketMat = dfhack.matinfo.find("PLANT_MAT:"..n.id..":WOOD")
+    end
+    if bucketMat then break end
+   end
+  end
+  local bucketType = dfhack.items.findType("BUCKET:NONE")
+  local bucket = df.item.find(dfhack.items.createItem(bucketType, -1, bucketMat.type, bucketMat.index, creator))
+  dfhack.items.moveToContainer(item, bucket)
+  guidm.setCursorPos(creator.pos)
+  dfhack.run_command("spotclean")
```

Comment:
can you restore the cursor to its former position after running `spotclean`?

alternately, if you have the time, you could add a `--cursor` option to `spotclean` like so many other tools have for specifying the position without a cursor